src/Microsoft.ApiDesignGuidelines.Analyzers/Core/DoNotDecreaseInheritedMemberVisibility.Fixer.cs

```diff
@@ -1,36 +1,88 @@
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.CodeActions;
+using Analyzer.Utilities;
 
 namespace Microsoft.ApiDesignGuidelines.Analyzers
-{                              
+{
     /// <summary>
     /// CA2222: Do not decrease inherited member visibility
     /// </summary>
-    public abstract class DoNotDecreaseInheritedMemberVisibilityFixer : CodeFixProvider
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public sealed class DoNotDecreaseInheritedMemberVisibilityFixer : CodeFixProvider
     {
         public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(DoNotDecreaseInheritedMemberVisibilityAnalyzer.RuleId);
 
         public sealed override FixAllProvider GetFixAllProvider()
         {
-            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
             return WellKnownFixAllProviders.BatchFixer;
         }
 
         public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
-        {                              
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await new Task(() => { });
-            throw new NotImplementedException();
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var expression = root.FindNode(context.Span);
+
+            if (expression != null)
+            {
+                var title = MicrosoftApiDesignGuidelinesAnalyzersResources.DoNotDecreaseInheritedMemberVisibilityTitle;
+                context.RegisterCodeFix(CodeAction.Create(title, c => IncreaseVisibility(context.Document, expression, c), equivalenceKey: title), context.Diagnostics);
+            }
+        }
+
+        private async Task<Document> IncreaseVisibility(Document document, SyntaxNode node, CancellationToken cancellationToken)
+        {
+            var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            var generator = editor.Generator;
+            var model = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var declaration = editor.Generator.GetDeclaration(node);
+            var symbol = editor.SemanticModel.GetDeclaredSymbol(node);
+            var ancestorTypes = symbol?.ContainingType?.GetBaseTypes() ?? Enumerable.Empty<INamedTypeSymbol>();
+            var hiddenOrOverriddenMembers = ancestorTypes.SelectMany(t => t.GetMembers(symbol?.Name));
+
+            // Check if a public member was overridden
+            if (hiddenOrOverriddenMembers.Any(s => s.DeclaredAccessibility == Accessibility.Public))
+            {
+                var property = (symbol as IMethodSymbol)?.AssociatedSymbol as IPropertySymbol;
+                if (property != null)
+                {
```

Comment:
Ah, good point. Event accessors can't have explicit visibility specifications, so what I actually need to do here is not flag event accessors. Will make that fix and add a test.